### PR TITLE
Make the MaxConsumeCount  param configurable

### PR DESCRIPTION
**Pull request by kedlas**

### CODE REVIEW

**Comment by prochac** [comment]
Let's use `uint`, negative values are useless here.
Also the merge should be synchronised with our update, as we rely on the former default value.

**Comment by kedlas** [comment]
The merge sync is the reason why the default value is 3. I would prefer to have the default value 0 (disabled), but for that we must have all ourt consymers ready in advance. This change proposed does not change any configuration which we currently use.

File: consumer.go [hunk 1]
```diff
@@ -107,6 +108,7 @@ var defaultConsumerConfig = consumerConfig{
 	MaxParallelMessages:    1,
 	InvalidMessageCallback: func(context.Context, InvalidMessage, error) {},
 	Metrics:                noop.Meter{},
+	MaxConsumeCount:        3,
```
Comment (kedlas) [inline consumer.go:111]: @prochac here is the default set to 3 not to bring any BC change

Comment (kedlas) [reply consumer.go:111]: it just makes the whole thing configurable without any BC change

**Comment by prochac** [comment]
> The merge sync is the reason why the default value is 3. I would prefer to have the default value 0 (disabled), but for that we must have all ourt consymers ready in advance. This change proposed does not change any configuration which we currently use.

OK, let's migrate our code to explicitly use `3`, or better, make it configurable by env, and later we can make the breaking change, as it's only us using this module so far :D

File: consumer.go [hunk 2]
```diff
@@ -190,6 +192,16 @@ func WithHistoryLimit(d time.Duration) ConsumerOption {
 	}
 }
 
+// WithMaxConsumeCount sets the maximal number of times a message can be consumed before it is ignored.
+// When message causes OOM it could lead to infinite loop in the consumers.
```
Comment (prochac) [inline consumer.go:196]: Not just OOM, any crash. Uncaught panic from `nil` pointer, SIGKILL, etc.

File: consumer.go [hunk 3]
```diff
@@ -94,6 +92,9 @@ type consumerConfig struct {
 	// If not set, it will look for message in the whole table.
 	// You may set this value when using partitioned table to search just in partitions you are interested in
 	HistoryLimit time.Duration
+	// MaxConsumeCount is the maximal number of times a message can be consumed before it is ignored.
+	// This is a safety mechanism to prevent infinite loops when a message causes OOM errors
```
Comment (prochac) [inline consumer.go:96]: Again, not just OOM. Any crash.

**Review by prochac — CHANGES_REQUESTED** [review]

**Review by prochac — APPROVED** [review]

**Pull request merged**